google_sheets_pkg/read_sheet.go

```diff
@@ -28,42 +32,104 @@ func (s ReadSheet) Execute(ctx step.Context) (interface{}, error) {
 }
 
 func (ReadSheet) execute(input ReadSheetInput) (*ReadSheetOutput, error) {
-	conf, err := google.JWTConfigFromJSON([]byte(input.Credentials), spreadsheet.Scope)
+	if input.UseFirstRowAsFields && len(input.Fields) > 0 {
+		return nil, errors.New("If you specify UseFirstRowAsFields, then you can not specify Fields.")
+	}
+	if input.UseFirstRowAsFields == false && len(input.Fields) == 0 {
+		return nil, errors.New("You must either specify `Fields` or UseFirstRowAsFields = true")
+	}
+	if input.Credentials == "" {
+		return nil, errors.New("No credentials were provided. Please provide your google service account key using the `Credentials` input")
+	}
+
+	conf, err := google.JWTConfigFromJSON([]byte(input.Credentials), "https://www.googleapis.com/auth/spreadsheets.readonly")
 	if err != nil {
 		return nil, err
 	}
-	client := conf.Client(context.Background())
+	ctx := context.Background()
+	srv, err := sheets.NewService(ctx, option.WithHTTPClient(conf.Client(ctx)))
```

Comment:
This should probably use your new method `ValidateInputsAndGetConf`

Reply:
updated